narwhal/storage/src/lib.rs

```diff
@@ -2,9 +2,57 @@
 // SPDX-License-Identifier: Apache-2.0
 
 mod certificate_store;
+mod header_store;
 mod node_store;
+mod payload_store;
 mod proposer_store;
+mod vote_digest_store;
 
 pub use certificate_store::*;
+use dashmap::DashMap;
+pub use header_store::*;
 pub use node_store::*;
+pub use payload_store::*;
 pub use proposer_store::*;
+use std::collections::VecDeque;
+use std::hash::Hash;
+use std::sync::Arc;
+use tokio::sync::oneshot;
+use tokio::sync::oneshot::{Receiver, Sender};
+use tracing::warn;
+pub use vote_digest_store::*;
+
+// A simple pub/sub to notify subscribers when a value becomes available.
+#[derive(Clone)]
+struct NotifySubscribers<K: Eq + Hash + Clone, V: Clone> {
+    notify_subscribers: Arc<DashMap<K, VecDeque<Sender<V>>>>,
```

Comment:
optional: using a watch channel may avoid copying the value

Reply:
Well I mostly used it for silly fairness (first come first served) - but I guess could use just `Vec` yes

Reply:
> optional: using a watch channel may avoid copying the value

@mwtian I am looking to use @andll 's [proposal](https://github.com/MystenLabs/sui/pull/9387#discussion_r1139296355) for `NotifyRead` - so probably the `NotfiySubscribers` will get replaced in follow up PR.